docs/cryptodoc/src/07_rng.rst

```diff
@@ -1025,34 +1025,64 @@ Random Number Generation for Probabilistic Public Key Algorithms
-It works as follows:
+Note that this algorithm allows to generate random numbers from an arbitrary range
+between a :math:`min` and a :math:`max` (exclusive) value. Typically, in public key algorithms,
+the :math:`min` parameter will be zero.
 
-.. admonition:: ``BigInt::random_integer()``
+The algorithm has a special case for :math:`min > 1` to avoid unpredictable runtimes
+if :math:`min` and :math:`max` are both large but their difference is small.
+For clarity, we describe the two cases independently:
 
-   **Input:**
+.. admonition:: ``BigInt::random_integer()`` (for :math:`min \in \{0,1\}`)
+
+   **Input**
 
-   -  ``rng``: random number generator (see :ref:`rng/main` for available algorithms)
-   -  ``min``: integer lower bound of desired range
-   -  ``max``: integer upper bound of desired range, excluding ``max``
+   -  :math:`rng`: random number generator (see :ref:`rng/main` for available algorithms)
+   -  :math:`min`: either 0 or 1
+   -  :math:`max`: integer upper bound of desired range, excluding :math:`max`
 
    **Output:**
 
-   -  ``r``: :math:`min \leq r < max`
+   -  :math:`r`: :math:`min \leq r < max`
 
    **Steps:**
 
-   1. Preliminary parameter requirement checks are conducted. ``min`` must be
-      smaller than ``max``, and neither can be negative.
-   2. Retrieve the bit length ``n`` of the ``max`` value.
-   3. Use ``rng`` to generate :math:`r \in {\{{0,...,{2^{n} - 1}}\}}`.
-   4. if (:math:`min \leq r < max`) return ``r``.
+   1. Preliminary parameter requirement checks are conducted. :math:`min` must be
+      smaller than :math:`max`, and neither can be negative.
+   2. Retrieve the bit length :math:`n` of the :math:`max` value.
+   3. Use :math:`rng` to generate :math:`r \in {\{{0,...,{2^{n} - 1}}\}}`.
+   4. If (:math:`min \leq r < max`) return :math:`r`.
    5. Go to Step 3.
 
-**Conclusion:** The algorithm is a slight adaptation of Algorithm B.1 in Section B.4 of [TR-02102-1]_
+**Note:** Technically, the base implementation above should be defined for exactly :math:`min = 0`
+not :math:`min \in \{0,1\}`. When `adding this special case in Botan 3.1.0 <https://github.com/randombit/botan/pull/3594>`_
+this was done to avoid breaking existing test cases that depended on the determinism of this method for :math:`min = 1`.
+
+.. admonition:: ``BigInt::random_integer()`` (for :math:`min > 1`)
+
+   **Input:**
+
+   -  :math:`rng`: random number generator (see :ref:`rng/main` for available algorithms)
+   -  :math:`min`: integer lower bound of desired range (with :math:`min > 1`)
+   -  :math:`max`: integer upper bound of desired range, excluding :math:`max`
+
+   **Output:**
+
+   -  :math:`r`: :math:`min \leq r < max`
+
+   **Steps:**
+
+   1. Preliminary parameter requirement checks are conducted. :math:`min` must be
+      smaller than :math:`max`, and neither can be negative.
+   2. Calculate :math:`diff` as the difference between :math:`max` and :math:`min`
```

Comment:
_diff_ in math font looks weird, but I don't know why. It's okay, I guess...

![image](https://github.com/sehlen-bsi/botan-docs/assets/126883116/034b27c7-152e-4803-a477-9e2db7f46b25)

Reply:
It is more by chance I think that the rest looks ok. I think in math mode it is more assumed that each letter is it's own variable. All multi letter variable names need probably to be in a `\mathit` or `\mathrm`.

See also https://tex.stackexchange.com/questions/482743/typesetting-multi-letter-variable-names-in-math-mode

Reply:
Yikes! That makes sense, I guess. I'll see what I can do.

Reply:
Mhh, I didn't find a convenient way to express this in Sphinx. I guess, for now, we have to live with this crappy kerning. 😞
One day we should donate a crate of beer for the one willing to go through the whole document and rectify all the formatting quirks.